src/Sentry/TransactionTracer.cs

```diff
@@ -362,6 +363,11 @@ public void Clear()
     /// <inheritdoc />
     public void Finish()
     {
+        if (_hasFinished.Exchange(true))
+        {
+            return;
```

Comment:
question: Timer

~Should we, in the case of a Timer still being active, disable/dispose the timer too when the transactions is transitioning from unfinished to finished?~

Wait ... I read this wrong ... this is actually what is happening:
- the first `Finish` will Stop/Dispose the Timer, if active
- subsequent `Finish`es will early-out, but the Timer, if active, has been stopped/disposed previously